src/ucp/core/ucp_context.c

```diff
@@ -1918,6 +1930,28 @@ static double ucp_context_get_memcpy_bw()
     return ucp_context_est_bcopy_bw[ucs_arch_get_cpu_vendor()];
 }
 
+static int
+ucp_dynamic_tl_switch_config_valid(const ucp_context_config_t *config)
+{
+    const char *err_str;
+
+    if (config->dynamic_tl_switch_interval == 0) {
+        err_str = "UCX_DYNAMIC_TL_SWITCH_INTERVAL";
+        goto err;
```

Comment:
IMO it would be simpler to print error message and exit, without goto
```c
    if (config->dynamic_tl_switch_interval == 0) {
        ucs_error("UCX_DYNAMIC_TL_SWITCH_INTERVAL must be > 0");
        return 0;
    }
```